src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -504,6 +504,50 @@ public void Reset()
             }
         }
 
+        public override long AvailableFreeSpace
+        {
+            get
+            {
+                return Quota - UsedSize;
+            }
+        }
+
+        [CLSCompliant(false)]
+        [Obsolete("IsolatedStorage.MaximumSize has been deprecated because it is not CLS Compliant.  To get the maximum size use IsolatedStorage.Quota")]
+        public override ulong MaximumSize
+        {
+            get
+            {
+                return long.MaxValue;
+            }
+        }
+
+        public override long Quota
+        {
+            get
+            {
+                return long.MaxValue;
+            }
+        }
+
+        public override long UsedSize
+        {
+            get
+            {
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_IsolatedStorageSize);
```

Comment:
I think if we're not going to have Quota throw we should probably do 0 for Current/UsedSize.